File: Projects/07/proj7.rb
def print_array(arr)
    arr.each { |elem| puts elem }
  end
  
  def read_array(size)
    arr = []
    size.times do |i|
      print "Enter a value for element #{i}: "
      arr << gets.chomp.to_f
    end
    arr
  end
  
  print "How many values do you want in the array? "
  num_values = gets.chomp.to_i
  
  arr = read_array(num_values)
  print_array(arr)

File: Projects/07/proj7.java
import java.util.Scanner;

public class proj7 {
    public static void printArray(double[] arr) {
        for (double elem : arr) {
            System.out.println(elem);
        }
    }

    public static void readArray(double[] arr) {
        Scanner sc = new Scanner(System.in);
        for (int i = 0; i < arr.length; i++) {
            System.out.print("Enter a value for element " + i + ": ");
            arr[i] = sc.nextDouble();
        }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("How many values do you want in the array? ");
        int numValues = sc.nextInt();

        double[] arr = new double[numValues];

        readArray(arr);
        printArray(arr);
    }
}